Source/WTF/wtf/StackTrace.cpp

```diff
@@ -45,6 +45,55 @@ void WTFGetBacktrace(void** stack, int* size)
 
 namespace WTF {
 
+#if USE(LIBBACKTRACE)
+static struct backtrace_state* backtraceState()
+{
+    static struct backtrace_state* backtraceState = backtrace_create_state(nullptr, 1, nullptr, nullptr);
+    return backtraceState;
+}
+
+static void backtraceSyminfoCallback(void* data, uintptr_t, const char* symname, uintptr_t, uintptr_t)
+{
+    const char** symbol = static_cast<const char**>(data);
+    *symbol = symname;
+}
+
+static int backtraceFullCallback(void* data, uintptr_t, const char*, int, const char* function)
+{
+    const char** symbol = static_cast<const char**>(data);
+    *symbol = function;
+    return 0;
+}
+
+const char** symbolize(void* const* addresses, int size)
+{
+    struct backtrace_state* state = backtraceState();
+    if (!state)
+        return nullptr;
+
+    const char** symbols = (const char**)malloc(sizeof(const char*) * size);
+
+    for (int i = 0; i < size; ++i) {
+        uintptr_t pc = reinterpret_cast<uintptr_t>(addresses[i]);
+        const char* symbol;
+
+        backtrace_pcinfo(state, pc, backtraceFullCallback, nullptr, &symbol);
+        if (!symbol)
+            backtrace_syminfo(backtraceState(), pc, backtraceSyminfoCallback, nullptr, &symbol);
+
+        if (symbol) {
+            char* demangled = abi::__cxa_demangle(symbol, nullptr, nullptr, nullptr);
```

Comment:
I see this is a libstdc++ extension. Is it supported by libc++?

This seems sufficiently nonportable that normally I'd like to see a feature test for it, but in this case I guess it's probably OK because libbacktrace is inherently specific to GCC and Clang. That said, please do make sure it won't break with libc++.

Reply:
I don't know how to test it with `libc++`. Any suggestions?

Reply:
Try building with clang and use `-stdlib=libc++` in `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS`. If it works then you're good.

Reply:
I searched through the [libc++ source code](https://github.com/llvm/llvm-project/tree/main/libcxx) and found a bunch of references to `__cxa_demangle`. It's implemented; you're fine.